library/core/src/str/mod.rs

```diff
@@ -658,13 +659,7 @@ impl str {
     #[must_use]
     #[stable(feature = "str_split_at", since = "1.4.0")]
     pub fn split_at(&self, mid: usize) -> (&str, &str) {
-        // is_char_boundary checks that the index is in [0, .len()]
-        if self.is_char_boundary(mid) {
-            // SAFETY: just checked that `mid` is on a char boundary.
-            unsafe { (self.get_unchecked(0..mid), self.get_unchecked(mid..self.len())) }
-        } else {
-            slice_error_fail(self, 0, mid)
-        }
+        self.split_at_checked(mid).unwrap_or_else(|| slice_error_fail(self, 0, mid))
```

Comment:
Could you replace the `unwrap_or_else` with `match` / `let Some(...) else ...`? Avoiding closures in `library` is ever so slightly better for compile times